benchmarks/src/main/java/io/druid/benchmark/ExpressionBenchmark.java

```diff
@@ -0,0 +1,249 @@
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+@State(Scope.Benchmark)
+@Fork(jvmArgsPrepend = "-server", value = 1)
+@Warmup(iterations = 15)
+@Measurement(iterations = 30)
+public class ExpressionBenchmark
+{
+  @Param({"1000000"})
+  private int rowsPerSegment;
+
+  private SegmentGenerator segmentGenerator;
+  private QueryableIndex index;
+  private JavaScriptAggregatorFactory javaScriptAggregatorFactory;
+  private DoubleSumAggregatorFactory expressionAggregatorFactory;
+  private ByteBuffer aggregationBuffer = ByteBuffer.allocate(Doubles.BYTES);
+
+  @Setup(Level.Trial)
+  public void setup() throws Exception
+  {
+    final BenchmarkSchemaInfo schemaInfo = new BenchmarkSchemaInfo(
+        ImmutableList.of(
+            BenchmarkColumnSchema.makeNormal("x", ValueType.FLOAT, false, 1, 0d, 0d, 10000d, false),
+            BenchmarkColumnSchema.makeNormal("y", ValueType.FLOAT, false, 1, 0d, 0d, 10000d, false)
+        ),
+        ImmutableList.of(),
+        new Interval("2000/P1D"),
+        false
+    );
+
+    final DataSegment dataSegment = DataSegment.builder()
+                                               .dataSource("foo")
+                                               .interval(schemaInfo.getDataInterval())
+                                               .version("1")
+                                               .shardSpec(new LinearShardSpec(0))
+                                               .build();
+
+    this.segmentGenerator = new SegmentGenerator();
+    this.index = segmentGenerator.generate(dataSegment, schemaInfo, rowsPerSegment);
+    this.javaScriptAggregatorFactory = new JavaScriptAggregatorFactory(
+        "name",
+        ImmutableList.of("x", "y"),
+        "function(current,x,y) { if (x > 0) { return current + x + 1 } else { return current + y + 1 } }",
+        "function() { return 0 }",
+        "function(a,b) { return a + b }",
+        JavaScriptConfig.getEnabledInstance()
+    );
+    this.expressionAggregatorFactory = new DoubleSumAggregatorFactory(
+        "name",
+        null,
+        "if(x>0,1.0+x,y+1)"
+    );
+  }
+
+  @TearDown(Level.Trial)
+  public void tearDown() throws Exception
+  {
+    if (index != null) {
+      index.close();
+      index = null;
+    }
+
+    if (segmentGenerator != null) {
+      segmentGenerator.close();
+      segmentGenerator = null;
+    }
+  }
+
+  @Benchmark
+  @BenchmarkMode(Mode.AverageTime)
```

Comment:
Since all benchmarks in the class use the same mode and time unit, could be moved to the class header